Client/Frontend/Sync/BraveCore/BraveCoreMigrator.swift

```diff
@@ -0,0 +1,245 @@
+                    self.observer = nil
+
+                    BraveSyncAPI.shared.leaveSyncGroup()
+                    self.bookmarksAPI.removeAll()
+                    self.migrate() { _ in
+                        didFinishTest = true
+                    }
+                }))
+            }
+        }
+        
+        while !didFinishTest {
+            RunLoop.current.run(mode: .default, before: .distantFuture)
+        }
+        
+        print("DONE TESTING MIGRATION")*/
+    }
+    
+    public func migrate(_ completion: ((_ success: Bool) -> Void)? = nil) {
+        if Preferences.Chromium.migrationBookmarks.value {
+            completion?(true)
+            return
+        }
+        
+        func performMigrationIfNeeded(_ completion: ((Bool) -> Void)?) {
+            if !Preferences.Chromium.migrationBookmarks.value {
+                log.info("Migrating to Chromium Bookmarks v1 - Exporting")
+                self.exportBookmarks { [weak self] success in
+                    if success {
+                        log.info("Migrating to Chromium Bookmarks v1 - Start")
+                        self?.migrateBookmarks() { success in
+                            Preferences.Chromium.migrationBookmarks.value = success
+                            completion?(success)
+                        }
+                    } else {
+                        log.info("Migrating to Chromium Bookmarks v1 failed: Exporting")
+                        completion?(success)
+                    }
+                }
+            } else {
+                completion?(true)
+            }
+        }
+        
+        //If the bookmark model has already loaded, the observer does NOT get called!
+        //Therefore we should continue to migrate the bookmarks
+        if bookmarksAPI.isLoaded {
+            performMigrationIfNeeded(completion)
+        } else {
+            //Wait for the bookmark model to load before we attempt to perform migration!
+            self.observer = bookmarksAPI.add(BookmarksModelLoadedObserver({ [weak self] in
+                guard let self = self else { return }
+                self.observer?.destroy()
+                self.observer = nil
+
+                performMigrationIfNeeded(completion)
+            }))
+        }
+    }
+    
+    public func exportBookmarks(to url: URL, _ completion: @escaping (_ success: Bool) -> Void) {
+        self.dataImportExporter.exportBookmarks(to: url, bookmarks: Bookmark.getAllTopLevelBookmarks()) { success in
+            completion(success)
+        }
+    }
+    
+    private func exportBookmarks(_ completion: @escaping (_ success: Bool) -> Void) {
+        let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
+        guard let documentsDirectory = paths.first else {
+            return completion(false)
+        }
+        
+        guard let url = URL(string: "\(documentsDirectory)/Bookmarks.html") else {
```

Comment:
I saw this url in few places, what about extracting it to a constant somewhere? to guarantee each call uses the same url